ibis/backends/polars/compiler.py

```diff
@@ -86,10 +86,14 @@ def _make_duration(value, dtype):
 def literal(op, **_):
     value = op.value
     dtype = op.dtype
+    if dtype.is_interval():
+        return _make_duration(value, dtype)
 
-    if dtype.is_array():
```

Comment:
Can you avoid moving the code around here and **only** handle the `None` case? It's harder to review when these kinds of changes are mixed together.

Reply:
Need to look at this again, but IIRC this was actually the minimum possible change, something was erroring with NULL duration handling. Will check again

Reply:
Yes, this reorg is needed. ibis's Duration.DAY and friends are not representable in polars' datatypes, so if you call `PolarsType.from_ibis(dtype)` on those, you get an error. So we need an early guard to look for the interval dtypes and return early.